api/client/build.go

```diff
@@ -296,6 +306,10 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		headers:     headers,
 	}
 
+	if *suppressOutput {
+		sopts.rawTerminal = false
+		sopts.out = bytes.NewBuffer(nil)
```

Comment:
Unfortunately, using `outBuf` here wouldn't work.
`outBuf` is used also in the `body` and in the `getContextURL's` progressbars.
`sopts.out` is a `io.Writer` and because of that when the build is successful, the output in line 343 will be the image ID with verbose progress logs.

I think it's still can be done using `bufio`. Let me see.